uiplib/scrape.py

```diff
@@ -118,56 +126,38 @@ def get_desktoppr_image_links(url, no_of_images):
     for result in responses:
         image_url = result['image']['url']
         filename = image_url.split('/')[-1]
-        image_links.append((filename, image_url))
+        image_links.append({
+                            'name': filename,
+                            'image_url': image_url
+                           })
 
     return image_links
 
 
-def get_image_links(url, count):
+def get_image_data_list(websites):
     """Return the links for images."""
     image_links = []
+    for url in websites:
+        if 'unsplash' in url:  # For Unsplash
+            image_links.extend(get_unsplash_image_links(url))
 
-    if 'unsplash' in url:  # For Unsplash
-        image_links.extend(get_unsplash_image_links(url, count))
-
-    elif 'reddit' in url:  # For Reddit
-        image_links.extend(get_reddit_image_links(url, count))
+        elif 'reddit' in url:  # For Reddit
+            image_links.extend(get_reddit_image_links(url))
 
-    elif 'desktoppr' in url:  # For Desktoppr
-        image_links.extend(get_desktoppr_image_links(url, count))
+        elif 'desktoppr' in url:  # For Desktoppr
+            image_links.extend(get_desktoppr_image_links(url))
 
     return image_links
 
 
-def download_store_images(full_path, image_link):
+def download_and_store_image(directory, image_data):
     """Download and store the images to the specified path."""
     try:
-        urlretrieve(image_link,
-                    full_path,
+        urlretrieve(image_data.get('image_url'),
```

Comment:
I think it’s better switching to requests library.

Reply:
that's another issue